build.zig

```diff
@@ -4,13 +4,24 @@ const builtin = @import("builtin");
 pub fn build(b: *std.build.Builder) void {
     const target = b.standardTargetOptions(.{});
     const mode = b.standardReleaseOptions();
+    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
+    const allocator = gpa.allocator();
 
     {
         const tigerbeetle = b.addExecutable("tigerbeetle", "src/main.zig");
         tigerbeetle.setTarget(target);
         tigerbeetle.setBuildMode(mode);
         tigerbeetle.install();
 
+        // The "tigerbeetle version" command includes the build-time commit hash.
+        const options = b.addOptions();
+        if (git_commit(allocator)) |commit| {
+            options.addOption(?[]const u8, "git_commit", commit[0..]);
+        } else {
+            options.addOption(?[]const u8, "git_commit", null);
+        }
+        tigerbeetle.addOptions("tigerbeetle_build_options", options);
```

Comment:
Woah, I did not know about this.

Reply:
:laughing: It isn't documented anywhere I could find. I'm thinking this is also how we will support comptime configuration from the user (rather than what we do right now, which is just manually modifying `config.zig`).

Zig has a couple of open issues that will make this even easier from the source code side: https://github.com/ziglang/zig/issues/1646 and https://github.com/ziglang/zig/issues/226.